### PR TITLE
Add camel support for erc165

### PR DESCRIPTION
**Pull request by andrew-fleming**
Fixes (partial) #596 

#### PR Checklist

- [x] Tests
- [ ] Tried the feature on a public network
- [ ] Documentation

### CODE REVIEW

**Comment by andrew-fleming** [comment]
This PR proposes to:
- Add camelCase tests for ERC165 in `test_erc165`. Maybe it's overkill? We can remove them if preferred
- Add an `erc165/` directory within `introspection/` to help keep things organized
- Add an `introspection/` directory inside of `tests/`
  - Again, this should help organization as opposed to having a long list of `test_<contract>`s

Also note, this PR will conflict with #629. It might be easier to merge that first and fix conflicts in this PR after.

**Comment by andrew-fleming** [comment]
Waiting on #637 to be merged first

File: src/openzeppelin/introspection/erc165/dual165.cairo [hunk 1]
```diff
@@ -23,8 +23,8 @@ trait DualCaseERC165Trait {
 
 impl DualCaseERC165Impl of DualCaseERC165Trait {
     fn supports_interface(self: @DualCaseERC165, interface_id: u32) -> bool {
-        let snake_selector = constants::SUPPORTS_INTERFACE_SELECTOR;
-        let camel_selector = constants::SUPPORTSINTERFACE_SELECTOR;
+        let snake_selector = constants::supports_interface_SELECTOR;
+        let camel_selector = constants::supportsInterface_SELECTOR;
```
Comment (ericnordelo) [inline src/openzeppelin/introspection/erc165/dual165.cairo:27]: Good readability improvement!

**Comment by martriay** [comment]
I think this PR got superseded by #637

**Pull request closed without merging**